docs/spikes/opencode-cli-format-json-2026-05-04.md

```
@@ -0,0 +1,135 @@
+| `step_finish` | `part.reason`, `part.cost`, `part.tokens` | Step ends — `reason: "stop"` is terminal |
+| `error` | `error.name`, `error.data.message` | Session error |
+
+**Final assistant message extraction (jq):**
+```bash
+ASSISTANT_TEXT=$(jq -r 'select(.type=="text") | .part.text' "$OUTPUT_FILE" | tr -d '\000')
+```
+
+Concatenate all `text` events for the full response. Multiple `text` events can be emitted per turn (streaming chunks).
+
+**Session ID extraction (jq):**
+```bash
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)